### DISCUSSION TITLE
How to enforce maximum length for MQTT topics

### DISCUSSION THREAD

**Discussion by haripriyapendem** [emqx/emqx#11643] [Q&A]
Currently MQTT maximum topic length by default set to 65535 bytes. https://github.com/emqx/emqx/issues/3727.

How do you enforce shorter length topic names. Is there any way to configure this.
I want to enforce topic length = 128 bytes and deny the access if topic length is more than 128 bytes.

**Comment by SergeTupchiy** [comment]
Hi @haripriyapendem,

There is no option to configure topic length limit. It's indeed limited to 65535 bytes according to MQTT spec.
However, you can implement a custom plugin that will do the necessary topic length check and deny access if your length limit is exceeded.
Documentation: https://docs.emqx.com/en/enterprise/v5.2/extensions/plugins.html#plugins
A good plugin example: https://github.com/zmstone/emqx_simple_acl

**Reply by haripriyapendem** [reply]
Thank you. I will try custom implementation
